src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Telemetry/SourceTelemetry.cs

```diff
@@ -0,0 +1,202 @@
+                return Path.Combine(programFiles, "Microsoft SDKs", "NuGetPackages");
+            }
+            catch
+            {
+                // Ignore this check if we fail for any reason to generate the path.
+                return null;
+            }
+        });
+
+        public static TelemetryEvent GetRestoreSourceSummaryEvent(
+            Guid parentId,
+            IEnumerable<PackageSource> packageSources)
+        {
+            return GetSourceSummaryEvent(
+                "RestorePackageSourceSummary",
+                parentId,
+                packageSources);
+        }
+
+        public static TelemetryEvent GetSearchSourceSummaryEvent(
+            Guid parentId,
+            IEnumerable<PackageSource> packageSources)
+        {
+            return GetSourceSummaryEvent(
+                "SearchPackageSourceSummary",
+                parentId,
+                packageSources);
+        }
+
+        /// <summary>
+        /// Create a SourceSummaryEvent event with counts of local vs http and v2 vs v3 feeds.
+        /// </summary>
+        private static TelemetryEvent GetSourceSummaryEvent(
+            string eventName,
+            Guid parentId,
+            IEnumerable<PackageSource> packageSources)
+        {
+            var local = 0;
+            var httpV2 = 0;
+            var httpV3 = 0;
+            var nugetOrg = HttpStyle.NotPresent;
+            var vsOfflinePackages = false;
+            var dotnetCuratedFeed = false;
+
+            if (packageSources != null)
+            {
+                foreach (var source in packageSources)
+                {
+                    // Ignore disabled sources
+                    if (source.IsEnabled)
+                    {
+                        if (source.IsHttp)
+                        {
+                            if (IsHttpV3(source))
+                            {
+                                // Http V3 feed
+                                httpV3++;
+
+                                if (IsHttpNuGetOrgDomainOrSubdomain(source))
+                                {
+                                    nugetOrg |= HttpStyle.YesV3;
+                                }
+                            }
+                            else
+                            {
+                                // Http V2 feed
+                                httpV2++;
+
+                                if (IsHttpNuGetOrgDomainOrSubdomain(source))
+                                {
+                                    if (source.Source.IndexOf(
+                                        "api/v2/curated-feeds/microsoftdotnet",
+                                        StringComparison.OrdinalIgnoreCase) >= 0)
+                                    {
+                                        dotnetCuratedFeed = true;
+                                    }
+                                    else
```

Comment:
Because `YesV2` is the main V2 feed on NuGet.org. The `microsoftdotnet` curated feed is something else (a subset). Additionally, depending on requirements of the person using the data, they could imply this.

Reply:
Gotcha. I would add a comment on the enum about this.